operations/helm/charts/mimir-distributed/templates/distributor/distributor-dep.yaml

```diff
@@ -47,10 +47,14 @@ spec:
             - "-target=distributor"
             - "-config.expand-env=true"
             - "-config.file=/etc/mimir/mimir.yaml"
-            # Force gRPC clients connecting to distributor to reconnect periodically in order to have them re-resolve endpoints and discover new replicas.
-            - "-server.grpc.keepalive.max-connection-age=2m"
+            # When write requests go through distributors via gRPC, we want gRPC clients to re-resolve the distributors DNS
+            # endpoint before the distributor process is terminated, in order to avoid any failures during graceful shutdown.
+            # To achieve it, we set a shutdown delay greater than the gRPC max connection age, and we set an even higher
+            # termination grace period (to give some extra buffer to the process to gracefully shutdown).
```

Comment:
[nit] Should we mention the termination grace period here, since it is actually not set in this yaml?

Reply:
You raised a very good point. I should have changed the termination grace period in Helm too. It wasn't detected by the linter because we currently skip the diff check for the termination grace period (I will try to fix it in a follow up PR).

Fixed in: https://github.com/grafana/mimir/pull/7361/commits/7c8a44c2141010e1adeabae56f1786e687ffe8d1

Reply:
Thank you @pracucci